RecoTracker/SpecialSeedGenerators/src/OutsideInMuonSeeder.cc

```diff
@@ -178,10 +184,19 @@ OutsideInMuonSeeder::produce(edm::Event & iEvent, const edm::EventSetup & iSetup
             }
         }
         if (tk.eta() > minEtaForTEC_) {
-            int iLayer = 9, found = 0;
-            std::vector< ForwardDetLayer const* > const & tec = measurementTracker->geometricSearchTracker()->posTecLayers();
-            for (auto it = tec.rbegin(), ed = tec.rend(); it != ed; ++it, --iLayer) {
-                if (debug_) std::cout << "\n ==== Trying TEC " << +iLayer << " ====" << std::endl;
+            auto forwLayers = measurementTracker->geometricSearchTracker()->posTecLayers();
+            if (tmpTkGeometry->isThere(GeomDetEnumerators::P2OTEC)) {
+              LogDebug("OutsideInMuonSeeder") << "\n We are using the Phase2 Outer Tracker (defined as a TID). ";
+              forwLayers = (measurementTracker->geometricSearchTracker()->posTidLayers());
+            }
+            LogTrace("OutsideInMuonSeeder") << "\n ==== TEC tot layers " << forwLayers.size() << " ====" << std::endl;
+            int found = 0;
+            int iLayer = forwLayers.size(); 
+            if(iLayer==0) LogError("OutsideInMuonSeeder") << "TEC+ has no layers." ;
+
+            if (debug_) std::cout << "\n ==== Tot layers " << forwLayers.size() << " ====" << std::endl;
```

Comment:
Why is there a mix between `LogDebug` and cout statements behind `debug_` flags? It seems like `LogDebug` should be used everywhere.